### PR TITLE
Shader internal mappings

### PR DESCRIPTION
**Pull request by Sollace**
Mapped the internals of the Shader class and also the class responsible for loading Mojang's custom "moj_" import statements.

### CODE REVIEW

File: mappings/GLImportProcessor.mapping [hunk 1]
```diff
@@ -0,0 +1,33 @@
+CLASS net/minecraft/class_5913 GLImportProcessor
```
Comment (liach) [inline mappings/GLImportProcessor.mapping]: Meh, this class has no package and  broke mappingpoet. I didn't expect that bad javadoc gen library to have this cool functionality!

**Review by liach — CHANGES_REQUESTED** [review]

File: mappings/GLImportProcessor.mapping [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	METHOD method_34232 parseImports (Ljava/lang/String;Lnet/minecraft/class_5913$class_5914;Ljava/lang/String;)Ljava/util/List;
+		ARG 1 source
+		ARG 2 context
+		ARG 3 path
+	METHOD method_34233 loadImport (ZLjava/lang/String;)Ljava/lang/String;
+		COMMENT Called to load an import reference's source code.
+		ARG 1 inline
+		ARG 2 name
+	CLASS class_5914 Context
+		COMMENT A context for the parser to keep track of it's current line and karet position in the file.
```
Comment (liach) [inline mappings/GLImportProcessor.mapping]:
```suggestion
		COMMENT A context for the parser to keep track of its current line and caret position in the file.
```

File: mappings/GLImportProcessor.mapping [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+		FIELD field_29202 columnNumber I
+		FIELD field_29203 lineNumber I
```
Comment (liach) [inline mappings/GLImportProcessor.mapping]:
```suggestion
		FIELD field_29202 column I
		FIELD field_29203 line I
```

File: mappings/GLImportProcessor.mapping (hunk 1 above)
Comment (modmuss50) [reply mappings/GLImportProcessor.mapping]: This has been an issue in mappingpoet for a while. We need another source of validation for classes not in a package.

File: mappings/GLImportProcessor.mapping [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+	COMMENT Handles the flattening of "moj_" import strings in the loaded GLSL shader file.
+	COMMENT Instances of an import are replaced by the contents of the referenced file
+	COMMENT prefixed by a comment describing the line position and original file location
+	COMMENT of the import.
+	FIELD field_29200 MOJ_IMPORT_PATTERN Ljava/util/regex/Pattern;
+	FIELD field_29201 IMPORT_VERSION_PATTERN Ljava/util/regex/Pattern;
+	METHOD method_34229 readSource (Ljava/lang/String;)Ljava/util/List;
+		COMMENT Reads the source code supplied into a list of lines suitable for uploading to
+		COMMENT the GL Shader cache.
+		COMMENT <p>Imports are processed as per the description of this class.</p>
```
Comment (liach) [inline mappings/GLImportProcessor.mapping]:
```suggestion
		COMMENT
		COMMENT <p>Imports are processed as per the description of this class.
```

File: mappings/GLImportProcessor.mapping (hunk 3 above)
Comment (Sollace) [reply mappings/GLImportProcessor.mapping]: It's a line _number_ not the line itself.

File: mappings/GLImportProcessor.mapping (hunk 4 above)
Comment (Sollace) [reply mappings/GLImportProcessor.mapping]: HTML should have a closing tag.

File: mappings/GLImportProcessor.mapping (hunk 1 above)
Comment (Sollace) [reply mappings/GLImportProcessor.mapping]: Oh, whoops! I forgot that I had to give it a package too. One moment.

File: mappings/net/minecraft/client/gl/GLImportProcessor.mapping [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+	COMMENT Handles the flattening of "moj_" import strings in the loaded GLSL shader file.
+	COMMENT Instances of an import are replaced by the contents of the referenced file
+	COMMENT prefixed by a comment describing the line position and original file location
+	COMMENT of the import.
+	FIELD field_29200 MOJ_IMPORT_PATTERN Ljava/util/regex/Pattern;
+	FIELD field_29201 IMPORT_VERSION_PATTERN Ljava/util/regex/Pattern;
+	METHOD method_34229 readSource (Ljava/lang/String;)Ljava/util/List;
+		COMMENT Reads the source code supplied into a list of lines suitable for uploading to
+		COMMENT the GL Shader cache.
+		COMMENT <p>Imports are processed as per the description of this class.</p>
```
Comment (liach) [inline mappings/net/minecraft/client/gl/GLImportProcessor.mapping]:
```suggestion
		COMMENT
		COMMENT <p>Imports are processed as per the description of this class.
```

File: mappings/net/minecraft/client/gl/GLImportProcessor.mapping [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+		ARG 2 context
+		ARG 3 path
+	METHOD method_34233 loadImport (ZLjava/lang/String;)Ljava/lang/String;
+		COMMENT Called to load an import reference's source code.
+		ARG 1 inline
+		ARG 2 name
+	CLASS class_5914 Context
+		COMMENT A context for the parser to keep track of its current line and caret position in the file.
+		FIELD field_29202 columnNumber I
+		FIELD field_29203 lineNumber I
```
Comment (liach) [inline mappings/net/minecraft/client/gl/GLImportProcessor.mapping]:
```suggestion
		FIELD field_29202 column I
		FIELD field_29203 line I
```

**Review by liach — APPROVED** [review]
Just stylistic stuff

**Comment by Sollace** [comment]
Rebased and updated to 21w13a

**Review by liach — APPROVED** [review]

File: mappings/net/minecraft/client/gl/GLImportProcessor.mapping [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+	FIELD field_29200 MOJ_IMPORT_PATTERN Ljava/util/regex/Pattern;
+	FIELD field_29201 IMPORT_VERSION_PATTERN Ljava/util/regex/Pattern;
+	METHOD method_34229 readSource (Ljava/lang/String;)Ljava/util/List;
+		COMMENT Reads the source code supplied into a list of lines suitable for uploading to
+		COMMENT the GL Shader cache.
+		COMMENT
+		COMMENT <p>Imports are processed as per the description of this class.
+		ARG 1 source
+	METHOD method_34230 readImport (Ljava/lang/String;I)Ljava/lang/String;
+		COMMENT
```
Comment (YanisBft) [inline mappings/net/minecraft/client/gl/GLImportProcessor.mapping]:
```suggestion
```

**Review by YanisBft — APPROVED** [review]

**Pull request merged**